index.d.ts

```diff
@@ -1,99 +1,487 @@
+   */
+  static formatAndSet(domElement: HTMLElement, options: Options): string;
+
+  /**
+   * Return the AutoNumeric object that manages the given DOM element
+   */
+  static getAutoNumericElement(domElement: HTMLElement): Input;
 
   /**
-   * Set the value, but do not save the new state in the history table (used for undo/redo actions)
+   * Return the default autoNumeric settings
+   */
+  static getDefaultConfig(): Options;
+
+  /**
+   * Return all the predefined options in one object
+   */
+  static getPredefinedOptions(): PredefinedOptions;
+
+  /**
+   * Return true if the given DOM element has an AutoNumeric object that manages it.
+   */
+  static isManagedByAutoNumeric(domElement: HTMLElement): boolean;
+
+  /**
+   * Unformat and localize the given formatted string with the given options.
+   */
+  static localize(value: string | HTMLElement, options: Options): string;
+
+  static localizeAndSet(domElement: HTMLElement, options: Options): string;
+
+  static mergeOptions(...options: Options[]): Options;
+
+  static reformatAndSet(referenceToTheDomElement: HTMLElement): void;
+
+  static test(domElement: HTMLElement): boolean;
+
+  static validate(options: Options): boolean;
+
+  static version(): string;
+
+  /**
+   * Set the given element value, and format it immediately.
+   * Additionally, this `set()` method can accept options that will be merged into the current AutoNumeric element, taking precedence over any previous settings.
+   *
+   * @example anElement.set('12345.67') // Formats the value
+   * @example anElement.set(12345.67) // Formats the value
+   * @example anElement.set(12345.67, { decimalCharacter : ',' }) // Update the settings and formats the value in one go
+   * @example anElement.northAmerican().set('$12,345.67') // Set an already formatted value (this does not _exactly_ respect the currency symbol/negative placements, but only remove all non-numbers characters, according to the ones given in the settings)
+   * @example anElement.set(null) // Set the rawValue and element value to `null`
+   *
+   * @param {number|string|null} newValue The value must be a Number, a numeric string or `null` (if `emptyInputBehavior` is set to `'null'`)
+   * @param {object} options A settings object that will override the current settings. Note: the update is done only if the `newValue` is defined.
+   * @param {boolean} saveChangeToHistory If set to `true`, then the change is recorded in the history table
+   * @returns {AutoNumeric}
+   * @throws
    */
   set(
     value: number | string | null,
-    options?: AutoNumericOptions,
-    save?: boolean
+    options?: Options,
+    saveChangeToHistory?: boolean
   ): void;
 
-  update(...options: AutoNumericOptions[]): void;
+  setUnformatted(value: number, options?: Options): void;
 
-  remove(): void;
+  get(callback?: (value: string, instance: AutoNumeric) => void): string;
 
+  getFormatted(
```

Comment:
Since the `get*` functions are one of the most important and used ones, it could help to add the jsdoc to those at least :
`Return the current formatted value of the AutoNumeric element as a string`